### PR TITLE
Minor enhancements to the JavaScript

### PR DESCRIPTION
**Pull request by hectorcorrea**
Made a few minor enhancements to the code to add creators including making sure they are rendered honoring their order and makes sure the "add me as creator" button uses an existing empty row (if any) instead of always adding a new row.

### CODE REVIEW

File: app/views/works/_edit_javascript.html.erb [hunk 1]
```diff
@@ -103,7 +164,12 @@
       var orcid = $("#user_orcid").val()
       var givenName = $("#user_given_name").val()
       var familyName = $("#user_family_name").val()
-      addCreatorHtml(num, orcid, givenName, familyName);
+      var emptyRowId = findEmptyCreator();
+      if (emptyRowId == null) {
+        addCreatorHtml(num, orcid, givenName, familyName);
+      } else {
+        setCreatorValues(emptyRowId, orcid, givenName, familyName)
```
Comment (hectorcorrea) [inline app/views/works/_edit_javascript.html.erb:171]: This is the enhancement...adds a new row for the "me" creator only if needed.

File: app/views/works/_edit_javascript.html.erb [hunk 2]
```diff
@@ -134,18 +200,12 @@
...
-      $(".creator-data").each(function(ix, el) {
-        var num = $(el).data("num");
-        var orcid = $(el).data("orcid");
-        var givenName = $(el).data("given-name");
-        var familyName = $(el).data("family-name");
-        var sequence = $(el).data("sequence");
-        addCreatorHtml(num, orcid, givenName, familyName, sequence);
-      });
+      // Adds the existing creators making sure we honor the ordering.
+      var creators = sortedCreators();
```
Comment (hectorcorrea) [inline app/views/works/_edit_javascript.html.erb:204]: Makes sure they are rendered based on their sequence on the database. They were rendering in order without this code but I wasn't sure they were always going to render in order so I decided to force them to be in order just to be sure.

File: app/views/works/_edit_javascript.html.erb [hunk 3]
```diff
@@ -48,18 +48,23 @@
...
+        if (emptyName) {
+          // delete it without asking
           $(rowToDelete).remove();
+        } else {
+          if (confirm(`Remove creator ${name}`)) {
+            $(rowToDelete).remove();
+          }
         }
+      } else {
+        console.log(`${rowToDelete} does not exist - nothing to do`);
```
Comment (jrgriffiniii) [inline app/views/works/_edit_javascript.html.erb:67]: I have no objection to leaving this here, but I just wish to ensure that this was not used solely for debugging.

File: app/views/works/_edit_javascript.html.erb [hunk 4]
```diff
@@ -92,6 +97,62 @@
...
+    var isEmptyRow = function(rowId) {
+      var selector, textboxes, i, textboxId, value;
+      selector = `#${rowId} > td > input`;
+      textboxes = $(selector);
+      for(i = 0; i < textboxes.length; i++) {
+        textboxId = textboxes[i].id;
+        if (textboxId.startsWith("orcid_") || textboxId.startsWith("given_name_") || textboxId.startsWith("family_name_")) {
+          value = $("#" + textboxId).val().trim();
+          if (value!= "") {
+            return false;
```
Comment (jrgriffiniii) [inline app/views/works/_edit_javascript.html.erb:129]: 👍

File: app/views/works/_edit_javascript.html.erb [hunk 5]
```diff
@@ -92,6 +97,62 @@
...
+      return true;
+    }
+
+    // Returns the ID of the first row that has an empty creator (if any)
+    var findEmptyCreator = function() {
+      var i;
+      var rows = $(".creators-table-row");
+      for(i = 0; i < rows.length; i++) {
+        if (isEmptyRow(rows[i].id)) {
+          return rows[i].id;
```
Comment (jrgriffiniii) [inline app/views/works/_edit_javascript.html.erb:142]: 👍

**Review by jrgriffiniii — APPROVED** [review]
Apologies, I left one question (https://github.com/pulibrary/pdc_describe/pull/172#discussion_r905072502), but this is otherwise excellent, and working very well for me in the `staging` environment. Thank you very much for this!

**Pull request merged**